runtime/src/mm/cpp/ExtraObjectData.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2010-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+#include "ExtraObjectData.hpp"
+
+#include "PointerBits.h"
+#include "Weak.h"
+
+#ifdef KONAN_OBJC_INTEROP
+#include "ObjCMMAPI.h"
+#endif
+
+using namespace kotlin;
+
+// static
+mm::ExtraObjectData& mm::ExtraObjectData::Install(ObjHeader* object) noexcept {
+    if (object->has_meta_object()) {
+        return mm::ExtraObjectData::FromMetaObjHeader(object->meta_object());
+    }
+
+    TypeInfo* typeInfo = object->typeInfoOrMeta_;
+    RuntimeCheck(!hasPointerBits(typeInfo, OBJECT_TAG_MASK), "Object must not be tagged");
+
+    auto* data = new ExtraObjectData(typeInfo);
```

Comment:
This doesn't use `konanAllocMemory`, does it?

Reply:
Nope, I treat it as "system data", so use the default allocator for it.

Reply:
I'll separately convert all the new code (and some old, using `std::make_unique` and `std::unique_ptr`) to the `konan*` stuff.

Reply:
https://github.com/JetBrains/kotlin-native/pull/4611